test/parallel/test-cli-eval.js

```diff
@@ -113,3 +113,16 @@ child.exec(nodejs + ` -e 'require("child_process").fork("${emptyFile}")'`,
   assert.strictEqual(proc.stderr, '');
   assert.strictEqual(proc.stdout, 'start\nbeforeExit\nexit\n');
 }
+
+[ '-arg1',
+  '-arg1 arg2 --arg3',
+  '--',
+].forEach(function(args) {
+  const evalopt = ' --eval "console.log(process.argv.slice(1).join(\' \'))"';
+  const cmd = `${nodejs}${evalopt} -- ${args}`;
+  child.exec(cmd, common.mustCall(function(err, stdout, stderr) {
+    assert.strictEqual(stdout, args + '\n');
+    assert.strictEqual(stderr, '');
+    assert.strictEqual(err, null);
+  }));
+});
```

Comment:
Before this PR:
_"--" is treated as an unknown node option. Hence it is always passed on to v8._

After this PR:
_"--" is treated as the end-of-options delimiter if and only if "--eval script" has been used prior to this option_

I can change it to do what @sam-github says it must do:
_"--" is treated as the end-of-options delimiter for all situations_

Since node also uses a script name to indicate end-of-options, I would like to clarify what should happen in this case?
`node --nodeopt1 script.js arg1 -- arg2`